Potential revoke commit fix

da-kami opened the pull request:
When working on #2272 I noticed that using the adaptor sig of the newly created commit tx here might not be correct.

Note on the second commit https://github.com/itchysats/itchysats/commit/e251fdddceb437d3517ffda62c1731ef753fc7de - This might not be backwards compatible so we might drop it.

Note that I think this might also be a breaking change for existing positions once we rollout rollover over libp2p! This might mean that positions that were opened prior to a release might get force closed. 

Note: I did *not* adapt this code in `setup_contract_deprecated`.

---

Note on breaking protocol changes:

We are currently not capable to distinguish different rollover versions per `Cfd`. We might want to add this feature by saving what rollover protocol version was used upon saving a rollover. Otherwise we accept that upon breaking change in the protocol *existing* positions that were opened prior to the breaking change might get force closed. I documented this in https://github.com/itchysats/itchysats/issues/2279

da-kami committed:
Use `dlc.commit` adaptor signature for revoke commit

We are revoking the previous commit, thus we should use the adaptor signature of the `commit` of the `dlc` and not the new one.

bonomat reviewed (approved):
I don't think this is a breaking change since we don't make use of this yet anyway 😅. 
@luckysori : what do you think about this? Is this correct now?

klochowicz reviewed (commented):
makes sense, based on my limited understanding of this part of the code.

luckysori commented on daemon/src/setup_contract.rs:
```diff
@@ -459,7 +459,7 @@ pub async fn roll_over(
         .try_into_msg2()?;
 
-    let revoked_commit = finalize_revoked_commits(&dlc, own_cfd_txs.commit.1, msg2.into())?;
+    let revoked_commit = finalize_revoked_commits(&dlc, dlc.commit.1, msg2.into())?;
```
:wrench: This `roll_over` function is not used anymore. Just remove it? 

Maybe you meant to modify the deprecated version? I wouldn't bother because it would be a breaking change for a feature that is not implemented fully. And the code should disappear soon.

da-kami replied:
Oh my, my effort to re-use code in that function was totally useless because it is unused 😅

I added a separate commit to remove the function. 
We might actually still adapt `contract_setup_deprecated` in that case - I thought that this was an even older version, but it is the code path used by the currently released takers. Given that I think we should update that code in a separate commit because it might be around for some more time until the takers upgraded.

luckysori reviewed (approved):
Amazing find! An 8-month old bug, a bit older than my daughter :baby:

I guess it would have been easier to find if we actually used and tested the revocation/punishment feature!

da-kami committed:
Remove unused `roll_over` function

When adapting this to re-use code parts we overlooked that this is actually not used.
Instead, we could opt for adapting the `setup_contract_deprecated.rs` version.
